web/packages/teleterm/src/ui/appContext.ts

```diff
@@ -150,6 +151,38 @@ export default class AppContext implements IAppContext {
 
   async init(): Promise<void> {
     this.setUpTshdEventSubscriptions();
+    this.mainProcessClient.subscribeToDeepLinkLaunch(result => {
+      if (result.status === 'error') {
+        let reason: string;
+        switch (result.reason) {
+          case 'unknown-protocol': {
+            reason = `The received URL is of an unknown protocol.`;
+            break;
+          }
+          case 'unsupported-uri': {
+            reason =
+              'The received URL does not point at a resource that can be launched from a deep link. ' +
+              'Either this version of Teleport Connect does not support launching this resource or the URL is incorrect.';
+            break;
+          }
+          case 'malformed-url': {
+            reason = `The URL of the deep link appears to be malformed.`;
+            break;
+          }
+          default: {
+            assertUnreachable(result);
+          }
+        }
+
+        this.notificationsService.notifyWarning({
+          title: 'Cannot open the deep link',
```

Comment:
would a user know what a deep link means? (i had to look it up). would it make more sense to say like `app link`? or just `link`?

Reply:
Good point. The distinction between a deep link and a regular link is not important for the end user, so I'm just going to change it to "link".